catalog/controller/extension/payment/razorpay.php

```diff
@@ -842,4 +867,97 @@ public function update()
             return  $this->response->redirect($this->url->link('extension/payment/razorpay/info', 'subscription_id=' . $postData['subscriptionId'], true));
         }
     }
+
+    protected function processSubscriptionAction($data)
+    {
+        $subscriptionId = $data['payload']['subscription']['entity']['id'];
+
+        if (empty($subscriptionId) === false) {
+            $merchant_order_id = $data['payload']['subscription']['entity']['notes']['merchant_order_id'];
+
+            if(isset($merchant_order_id) === true)
+            {
+                switch ($data['event']) {
+                    case 'subscription.paused':
+                        $status = 'paused';
+                        $oc_status = 2;
+                        break;
+
+                    case 'subscription.resumed':
+                        $status = 'active';
+                        $oc_status = 1;
+                        break;
+
+                    case 'subscription.cancelled':
+                        $status = 'cancelled';
+                        $oc_status = 3;
+                        break;
+                }
+
+                $this->load->model('extension/payment/razorpay');
```

Comment:
Do check if the status is already changed or not don't directly update the subscription status without checking the status